src/util.rs

```diff
@@ -472,7 +492,58 @@ pub(crate) mod ptr {
         }
     }
 
-    impl<'a, T: 'a + ?Sized> Debug for Ptr<'a, T> {
+    impl<'a, T: 'a + ?Sized, const ASSUME_ALIGNED: bool> Debug for Ptr<'a, T, ASSUME_ALIGNED> {
+        #[inline]
+        fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+            self.ptr.fmt(f)
+        }
+    }
+
+    /// A reference to a validly-initialized `T` of lifetime `'a` that might
+    /// *not* be validly aligned for `T`.
+    #[doc(hidden)]
+    #[allow(unreachable_pub)] // false positive?
+    pub struct MaybeAligned<'a, T: ?Sized> {
+        /// INVARIANTS:
+        /// 1. `ptr` is validly-initialized for `T`
```

Comment:
```suggestion
        /// INVARIANT: `*ptr` is validly-initialized for `T`
```

I assume you're not planning on adding other invariants? Also, it's not `ptr` that's initialized, but `*ptr` (you could also say "`ptr`'s referent", etc).